apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/SnapshotsExpirationSparkApp.java

```diff
@@ -18,41 +17,68 @@
 @Slf4j
 public class SnapshotsExpirationSparkApp extends BaseTableSparkApp {
   private final String granularity;
-  private final int count;
+  private final int maxAge;
+  private final int versions;
+
+  public static class DEFAULT_CONFIGURATION {
+    public static final int MAX_AGE = 3;
+    public static final String GRANULARITY = "DAYS";
+    public static final int VERSIONS = 0;
+  }
+
+  private static final String DEFAULT_GRANULARITY = "";
+
+  // By default do not define versions, and only retain snapshots based on max age
+  private static final String DEFAULT_VERSIONS = "0";
 
   public SnapshotsExpirationSparkApp(
-      String jobId, StateManager stateManager, String fqtn, String granularity, int count) {
+      String jobId,
+      StateManager stateManager,
+      String fqtn,
+      int maxAge,
+      String granularity,
+      int versions) {
     super(jobId, stateManager, fqtn);
-    this.granularity = granularity;
-    this.count = count;
+    if (maxAge == 0 && versions == 0) {
```

Comment:
What happens if maxAge and versions both are 0? Should we apply default policy in such case like expire 3 days worth of snapshot?

Reply:
I see that default configuration is set to 3 days. Sounds good.